### ISSUE TITLE
More general unit parsing

### ISSUE THREAD

**Issue by Kolaru** [PainterQubits/Unitful.jl#374]
Currently Unitful hability to understand units in non standard form is limited. So I built up a small utility to parse units from a more general form, eg

```julia
julia> destructure_units("nanoamps3 seconds / micrometer^2")
3-element Array{Any,1}:
 ("n", "A", 3)
 ("", "s", 1)
 ("μ", "m", -2)

julia> reduce_units_expr("nanoamps3 seconds / micrometer^2")
"nA^3*s*μm^-2"
```

The code can be found as an unregistered package at https://github.com/Kolaru/UnitsParser.jl. It basically just is a simple parser that look up for prefix and units aliases in a table and replace them with the corresponding symbol.

If there is interest here, I can turn it into a PR to Unitful. Otherwise I can also have it as a separate one.

**Comment by OliverEvans96** [comment]
I agree that this is a really nice feature! Thanks @Kolaru for uploading your implementation. One of the reasons I like using [Pint](https://pint.readthedocs.io/en/stable/) in Python is because of their incredibly flexible parsing. I think it would definitely be worth merging this into Unitful.
